File: tournament-winner/TournamentWinner.swift
import Foundation

class TournamentWinner {
    let AWAY_TEAM_WON: Int = 0
    let HOME_TEAM_WON: Int = 1
    let COUNT: Int = 3;
    public func tournamentWinner(_ competitions: [[String]], _ results: [Int]) -> String {
        var winner = ""
        var scores: [String: Int] = [:]
        for index in stride(from: 0, to: competitions.count, by: 1) {
            var winningTeam = competitions[index][results[index] == AWAY_TEAM_WON ? HOME_TEAM_WON : AWAY_TEAM_WON]
            if let score = scores[winningTeam] {
                scores[winningTeam] = score + COUNT
            } else {
                scores[winningTeam] = COUNT
            }
            winner = scores[winner, default: 0] < scores[winningTeam]! ? winningTeam : winner
        }
        return winner
    }
}

File: tournament-winner/TournamentWinner.cs
using System;
public class TournamentWinner {
    private readonly int AWAY_TEAM_WON = 0;
    private readonly int HOME_TEAM_WON = 1;
    private readonly int COUNT = 3;
    public string TournamentWinners(List<List<string>> competitions, List<string> results) {
        Dictionary<string, int> scores = new Dictionary<string, int>();
        string winner = "";
        for(int i = 0; i < competitions.Count; i++) {
            List<string> competition = competitions[i];
            string winningTeam = competition[results[i].Equals(AWAY_TEAM_WON) ? HOME_TEAM_WON : AWAY_TEAM_WON];
            scores[winningTeam] = scores.GetValueOrDefault(winningTeam) + COUNT;
            winner = scores.GetValueOrDefault(winner) < scores[winningTeam] ? winningTeam : winner;
        }
        return winner;
    }
}